charts/s3gw/templates/secret.yaml

```diff
@@ -1,3 +1,10 @@
+{{- $s3AccessKey := .Values.accessKey }}
+{{- $s3SecretKey := .Values.secretKey }}
+{{- if .Values.useExistingSecret }}
+{{- $keysData := (lookup "v1" "Secret" .Release.Namespace .Values.existingSecret).data }}
+{{- $s3AccessKey = b64dec (default "" $keysData.accessKey) }}
```

Comment:
are you suggesting to give the user the option to specify the names of the fields for the existing secret?

Reply:
If it is very obvious in the documentation that the secret must be set up this way, then it's ok.
I don't think namespaces are much of an issue, but if they are, we can still replace `.Release.Namespace` with a value from a user defined field, later on.

One thing that comes to my mind here is that these values are just used for initialization, right? So if a user supplies their custom secret, then deploys the s3gw, and then changes that custom secret, the s3gw won't pick up the change. This isn't a huge issue IMO, but should be mentioned in the documentation.
One way around that would be to require the user to supply `RGW_DEFAULT_USER_ACCESS_KEY` and `RGW_DEFAULT_USER_SECRET_KEY` as keys in their secret and then mount the user-supplied secret directly.

Reply:
> If it is very obvious in the documentation that the secret must be set up this way, then it's ok. 

ok, let's assume this will be very clear on doc.

>I don't think namespaces are much of an issue, but if they are, we can still replace `.Release.Namespace` with a value from a user defined field, later on.

ok
 
> One thing that comes to my mind here is that these values are just used for initialization, right? 

Yes, this secret is used on initialization phase; it is convenient when you are installing s3gw as subchart in another project.
This way, the embedder chart can also read from that secret and initialize all the S3 clients that will use s3gw.

> So if a user supplies their custom secret, then deploys the s3gw, and then changes that custom secret, the s3gw won't pick up the change. This isn't a huge issue IMO, but should be mentioned in the documentation. One way around that would be to require the user to supply `RGW_DEFAULT_USER_ACCESS_KEY` and `RGW_DEFAULT_USER_SECRET_KEY` as keys in their secret and then mount the user-supplied secret directly.

let's keep in mind that this is intended for initialization only.
`RGW_DEFAULT_USER_ACCESS_KEY` and `RGW_DEFAULT_USER_SECRET_KEY` are read by radosgw, but they are processed only if there is not already a default user in the system.
So, in the end, modifying those values won't change the default user credentials.
Moreover, this won't solve the problem that you still will have to change all the clients's configuration.

Reply:
In that case we don't lose any functionality, because we're using `RGW_DEFAULT_USER_ACCESS_KEY` and `RGW_DEFAULT_USER_SECRET_KEY` right now ourselves.
It should be made clear in the docs that these settings are used during initialization only. And I mean both our existing settings as well as this change.

But thanks for clarifying this.